sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```
@@ -560,6 +589,9 @@ class StreamExecution(
   @volatile private var noNewData = false
 
   override def processAllAvailable(): Unit = {
+    if (streamDeathCause != null) {
```

Just make `processAllAvailable` fail fast.